package proto;

// 服务器类型，新加都需要在填写
enum ServerType {
	SVR_TYPE_ROUTER = 0x1;		// 路由服務
	SVR_TYPE_CONN = 0x2;		// 客户端连接服务
	SVR_TYPE_TRANSFER = 0x3;	// 转发服务
}

message Ret {
	optional int32 code = 1;
	optional string what = 2;
}

// 新连接
message SocketClientIn {
}

// 连接断开
message SocketClientOut {
}

// 普通服务注册
message RegisterServerReq {
	optional int32 server_type = 1;
	optional int32 instance_id = 2;
	optional int32 server_zone = 3;
}

message RegisterServerRsp {
	optional Ret ret = 1;
}

// 普通服务心跳
message SvrHeatBeat {
	optional int32 server_type = 1;
	optional int32 instance_id = 2;
}

// transfer服务注册
message RegisterTransferReq {
	optional int32 server_type = 1;
	optional int32 instance_id = 2;
	optional int32 server_zone = 3;
}

message RegisterTransferRsp {
	optional Ret ret = 1;
}

// transfer服务心跳
message TransferHeatBeat {
	optional int32 server_zone = 1;
	optional int32 instance_id = 2;
}

// 更新login信息到transfer
message UpdateLoginInfoToTransfer {
	optional uint64 userid = 1;
	optional int32 server_zone = 2;
}